varinfo/umm_var.py

```diff
@@ -338,3 +341,53 @@ def get_json_serializable_value(input_value: Any) -> Any:
         output_value = input_value
 
     return output_value
+
+
+def publish_umm_var(collection_id: str,
+                    umm_var_dict: Dict,
+                    token: str,
+                    cmr_env: CMR_UAT) -> str:
```

Comment:
Yes, that is true! Thanks for catching this!

Reply:
I ended up doing this: `str|requests.Response:`

Reply:
Would it make sense to put the entire response for both success and failures and then check them after the fact?  I think you're going to find it awkward to work with both strings and json objects coming out of this function.

Reply:
I agree it has been awkward, but if I don't put it in this function I will have to put in `publish_all_umm_var` function and deal with the two different types there. The awkwardness comes from the fact that the types are different depending on if request passes or fails.

Reply:
Just following up, I was suggesting that you put the full response for both a success and a failure into the results, so that you can treat both the same where ever you end up handling them. so that you don't have to test the type to see if it's a success or failure. 

So that here https://github.com/nasa/earthdata-varinfo/blob/3529a6491ed524a03082a297efb5e922d7e76771/varinfo/umm_var.py#L377

You would return the full response in either case and then you would post-process by checking the `response.ok` value later.